test/functional/driver/driver-e2e-specs.js

```diff
@@ -293,6 +294,47 @@ describe('XCUITestDriver', function () {
         res.events.newSessionStarted[0].should.be.above(res.events.newSessionRequested[0]);
       });
     });
+    describe('w3c compliance', () => {
+      const sessionUrl = `http://${HOST}:${PORT}/wd/hub/session`;
+      it('should accept w3c formatted caps', async function () {
+        const { status, value, sessionId } = await request.post({url: sessionUrl, json: W3C_CAPS});
+        status.should.equal(0);
+        value.should.exist;
```

Comment:
java client requires the value to be a map, which must contain two keys of type string: 
`sessionId` and `capabilities `. Also all keys of `capabilities` map must be of type string.

See https://github.com/SeleniumHQ/selenium/blob/e0e8f01c508f17e6f80137cc6e020f7233f4899d/java/client/src/org/openqa/selenium/remote/W3CHandshakeResponse.java#L83 for more details